src/hotspot/cpu/s390/gc/shared/barrierSetAssembler_s390.hpp

```diff
@@ -66,6 +66,8 @@ class BarrierSetAssembler: public CHeapObj<mtGC> {
   OptoReg::Name refine_register(const Node* node,
                                 OptoReg::Name opto_reg) const;
 #endif // COMPILER2
+
+  static const int PATCHABLE_INSTRUCTION_OFFSET = 3 * 6 + 2;
```

Comment:
Maybe better call it `PATCHABLE_BARRIER_VALUE_OFFSET`?
It's no longer the instruction start address.
Or even better: define 2 constants. The version without the +2 is still needed below.